puppet.gemspec

```diff
@@ -40,5 +41,7 @@ Gem::Specification.new do |spec|
     # ffi 1.16.0 - 1.16.2 are broken on Windows
     spec.add_runtime_dependency('ffi', '>= 1.15.5', '< 1.17.0', '!= 1.16.0', '!= 1.16.1', '!= 1.16.2')
     spec.add_runtime_dependency('minitar', '~> 0.9')
+  else
+    spec.add_runtime_dependency('syslog', '~> 0.1.2') unless defined?(JRUBY_VERSION)
```

Comment:
I don't know if the `unless` is smart, I copied it from the http-cookie gem

Reply:
now it tries to install syslog on windows? Not sure why that's happening :thinking:

Reply:
Ah, the issue is that `Gem::Specification#platform` returns the platform for this spec, not the platform you're running on. The reason we do that is because we need to build platform-specific puppet gems on Linux like  https://github.com/puppetlabs/puppet/blob/d2f3cd52eecdca2243190a7c1ecf8fd55a02d143/Rakefile#L36
so that we can add platform-specific runtime dependencies like `ffi` when building for example https://rubygems.org/gems/puppet/versions/8.10.0-x64-mingw32

The reason you were seeing `syslog` being installed on windows, is because the default `Gem::Specification#platform` is `ruby`.

I'm thinking our `puppet.gemspec` may need to take into account the platform you're running on too.

Reply:
I'd suggest the following

```diff
diff --git a/Gemfile b/Gemfile
index 366b326c39..4ac7e39bf8 100644
--- a/Gemfile
+++ b/Gemfile
@@ -35,7 +35,6 @@ group(:features) do
   # requires native ldap headers/libs
   # gem 'ruby-ldap', '~> 0.9', require: false, platforms: [:ruby]
   gem 'puppetserver-ca', '~> 2.0', require: false
-  gem 'syslog', '~> 0.1.1', require: false, platforms: [:ruby]
   gem 'CFPropertyList', ['>= 3.0.6', '< 4'], require: false
 end

diff --git a/puppet.gemspec b/puppet.gemspec
index ed1944f596..47ed40911e 100644
--- a/puppet.gemspec
+++ b/puppet.gemspec
@@ -30,15 +30,18 @@ Gem::Specification.new do |spec|
   spec.add_runtime_dependency('puppet-resource_api', '~> 1.5')
   spec.add_runtime_dependency('scanf', '~> 1.0')
   spec.add_runtime_dependency('semantic_puppet', '~> 1.0')
+  spec.add_runtime_dependency('ostruct', '~> 0.6.0')

   platform = spec.platform.to_s
   if platform == 'universal-darwin'
     spec.add_runtime_dependency('CFPropertyList', ['>= 3.0.6', '< 4'])
   end

-  if platform == 'x64-mingw32' || platform == 'x86-mingw32'
+  if Gem::Platform::WINDOWS.include?(spec.platform) || Gem::Platform::WINDOWS.any? { |p| Gem::Platform.match(p) }
     # ffi 1.16.0 - 1.16.2 are broken on Windows
     spec.add_runtime_dependency('ffi', '>= 1.15.5', '< 1.17.0', '!= 1.16.0', '!= 1.16.1', '!= 1.16.2')
     spec.add_runtime_dependency('minitar', '~> 0.9')
+  else
+    spec.add_runtime_dependency('syslog', '~> 0.1.2')
   end
 end
```